python/sglang/srt/layers/rotary_embedding.py

```diff
@@ -222,17 +222,52 @@ def forward_cuda(
         query: torch.Tensor,
         key: torch.Tensor,
         offsets: Optional[torch.Tensor] = None,
+        layer: Any = None,  # RadixAttention
+        forward_batch=None,
```

Comment:
maybe we should not put layer,forward_batch etc into such a low-level. I personally suggest to put the k_buffer_ptr, v_buffer_ptr, etc, down to here. If too verbose, maybe make a simple @dataclass to pack them